src/app/_components/Event/EventFilter.js

```diff
@@ -0,0 +1,179 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useInView } from 'react-intersection-observer';
+import { useSearchParams } from 'next/navigation';
+import { CircularProgress } from '@mui/material';
+import CheckMark from '@icons/check-mark.svg';
+import Search from '@icons/search.svg';
+import { allEvents, useEventSetParam } from '@hooks';
+import { cn } from '@/utils/cn';
+import { PillButton } from '../PillButton';
+import { buttonColorVariant } from '../PillButton/style';
+import { NoInfoToShow } from '../NoInfoToShow';
+import { EventCard } from './Card';
+import { monthsAhead, monthNames } from './config';
+
+export function EventFilter() {
+  const [months, setMonths] = useState([]);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [activeMonth, setActiveMonth] = useState(months[0]);
+  const [activeMonthNumber, setActiveMonthNumber] = useState('');
```

Comment:
@Yivgha for me it looks like those 3 state are all the same. we can use only one state and others would be derived from it

Reply:
changed